spec/controllers/payment_gateways/stripe_controller_spec.rb

```diff
@@ -151,5 +151,120 @@ module PaymentGateways
         end
       end
     end
+
+    describe "#authorize" do
+      let(:customer) { create(:customer) }
+      let(:order) {
+        create(:order_with_totals, customer: customer, distributor: customer.enterprise,
+                                   state: "payment")
+      }
+      let(:payment_method) { create(:stripe_sca_payment_method) }
+      let!(:payment) {
+        create(
+          :payment,
+          payment_method: payment_method,
+          cvv_response_message: "https://stripe.com/redirect",
+          response_code: "pi_123",
+          order: order,
+          state: "requires_authorization"
+        )
+      }
+
+      before do
+        allow(controller).to receive(:spree_current_user) { current_user }
+      end
+
+      context "after returning from Stripe to authorize a payment" do
+        let(:current_user) { order.user }
+
+        context "with a valid payment intent" do
+          let(:payment_intent) { "pi_123" }
+          let(:payment_intent_response) { double(id: "pi_123", status: "requires_capture") }
+
+          before do
+            allow(Stripe::PaymentIntentValidator)
+              .to receive_message_chain(:new, :call).and_return(payment_intent_response)
+
+            allow(Spree::Order).to receive(:find_by) { order }
+          end
+
+          context "when the order is in payment state" do
+            it "completes the payment" do
+              expect(order).to receive(:process_payments!) do
+                payment.complete!
+              end
+
+              get :authorize, params: { order_number: order.number, payment_intent: payment_intent }
+
+              expect(response).to redirect_to order_path(order)
+              payment.reload
+              expect(payment.state).to eq("completed")
+              expect(payment.cvv_response_message).to be nil
+            end
+          end
+
+          context "when the order is already completed" do
+            before do
+              order.update_columns(state: "complete")
```

Comment:
I don't know if this `update_columns` is necessary here, but it break the linter.

Reply:
I forgot that one #2manyPR's